types/types.go

```diff
@@ -0,0 +1,73 @@
+package types
+
+import (
+	"log"
+	"sync"
+)
+
+// Runner is an interface which is implemented by all proxies
+type Runner interface {
+	Run() error
+	Cancel() error
+	Reload() error
+}
+
+// Server governs a slice of Runners
+type Server struct {
+	Servers []Runner
+	wg      sync.WaitGroup
+	Done    chan error
```

Comment:
maybe `Done()` should be part of the `Runner{}` interface. thoughts?

Reply:
Not sure